OKTA-727230 Add new event type (#4817)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/reference/api/itp-et/index.md

| Risk            | Contains the level of risk for a particular request (`LOW`, `MEDIUM`, or `HIGH`) and the `reasons` that contributed to the risk level.                | key-value pairs         |`{reasons=Anomalous Geo-Distance, New Device, New ASN, New IP, New State, New Country, New City, level=HIGH}`          |
| ThreatSuspected            | If ThreatInsight is running and detects a request as suspicious, the value for this property is `true`.                   | Boolean         | `false`         |
| TraceId            | A unique ID that is used across a single flow of ITP events to easily correlate them all into one System Log query                 | String         | `65d55fa6-b5a9-40f9-a6f1-627b9fa71b50`        |
| **target** (User)         | The user associated with the risk change           | Object     |        |
| type        | The type of target object     | String     | User       |

> | **target** (Rule)         | The rule associated with the continuous access evaluation            | Object     |        |

Author: Thanks, yeah, for 2024.06.0 we're not updating dev docs, but it's on the to-do list. I'll keep things the same for this pr. It will get updated. Thanks

| type        | The type of actor object           | String     | SecurityEventProvider        |

### user.authentication.universal_logout

**Description:** This event triggers when Okta or an admin invokes Universal Logout against an app instance. It contains the app instance details for which the Universal Logout API was triggered. The event indicates when apps have had Universal Logout triggered for audit or debugging purposes. This event is only triggered once. It's only triggered for apps that have been configured for Universal Logout. You can configure it in an Entity risk policy or Continuous Access, or invoke it manually from the user profile.

| Key event properties | Description                                         | Data type      | Example values |
| --------------------- | --------------------------------------------------- | -------------- | -------------- |
| **event.system.debugContext.debugData**                |                 |         |         |
| AppInstanceIds            |A list of application IDs that Okta triggered for Universal Logout                 | Array of IDs         | ["0oa1ysra5y0ESChAr0h8"]        |

> | TraceId            | The `TraceId` is used in continuous access evaluation use cases. A request that triggers a CAE evaluation can ultimately trigger things like CAE action events - and those are executed from the async jobs. `TraceId` connects together events triggered both by the original request handler and from the async jobs triggered by this handler. | String         | `94384405-51e3-4e13-b8b0-ba857b585a63`         |

Reviewer: We don't use the term "CAE" anywhere

Author: Thanks, did a search for all and replaced them.

| **target** (User)         | The user impacted by the universal logout          | Object     |        |
| type        | The type of target object     | String     | User       |
| **actor**                 |  The admin or system principal that triggers universal logout                 | Object        |         |
| type        | The type of actor object           |      |        |
| **client**                |  The client of the system principal actor for continuous access evaluation and entity risk policy actions, or the client of the admin triggering the clear user sessions action.                 | Object      |         |
| IPAddress              | IP address                |       |         |

### user.authentication.universal_logout.scheduled

> **Description:** This event triggers only when an admin manually triggers the Universal Logout against an app instance. It contains the location of the admin and the context of the universal logout, that is, from where and how the Universal Logout API was triggered. This event is only triggered once. You can co-relate this event with the `user.authentication.universal_logout` event using the `traceID` found under `DebugData` for both events.

Reviewer: ```suggestion
**Description:** This event triggers only when an admin manually triggers the Universal Logout against an app instance. It contains the location of the admin and the context of the Universal Logout, that is, from where and how the Universal Logout API was triggered. This event is only triggered once. You can co-relate this event with the `user.authentication.universal_logout` event using the `traceID` found under `DebugData` for both events.
```

Reviewer: Should "co-related" be correlate?

Author: Yes, that works far better.

| Key event properties | Description                                         | Data type      | Example values |
| --------------------- | --------------------------------------------------- | -------------- | -------------- |
| **event.system.debugContext.debugData**                |                 |         |         |

> | TraceId            | The `TraceId` is used in continuous access evaluation use cases. A request that triggers a CAE evaluation can ultimately trigger things like CAE action events - and those are executed from the async jobs. `TraceId` connects together events triggered both by the original request handler and from the async jobs triggered by this handler. | String         | `94384405-51e3-4e13-b8b0-ba857b585a63`         |

Reviewer: CAE

> | **target** (User)         | The user impacted by the universal logout          | Object     |        |

Reviewer: ```suggestion
| **target** (User)         | The user impacted by the Universal Logout          | Object     |        |
```

| type        | The type of target object     | String     | User       |

> | **actor**                 |  The admin or system principal that triggers universal logout                 | Object        |         |

Reviewer: ```suggestion
| **actor**                 |  The admin or system principal that triggers Universal Logout                 | Object        |         |
```

| Key event properties  | Description                                         | Data type      | Example values |
| --------------------- | --------------------------------------------------- | -------------- | -------------- |
| **event.system.debugContext.debugData**                |                 |         |         |
| EndedSessionId             |  The session ID that is ended for the target user                 | String         |  `idxffK-esRDSrC5m0ly-Kma9A `       |
| TraceId            | A unique ID that is used across a single flow of ITP events to easily correlate them all into one System Log query                  | String         | `e1214f29-e6b3-4698-b3be-4bccaadf1937`         |
| ThreatSuspected             |  If ThreatInsight is running and detects a request as suspicious, the value for this property is `true`.                   | Boolean         |          |

> | Url                        | The log-out URL from the end user or admin actor                  | String         |          |

Reviewer: ```suggestion
| Url                        | The logout URL from the end user or admin actor                  | String         |          |
```
